File: data_to_json.js
const fs = require("fs");
const path = require("path");

const baseDir = "./data_files";
let output = [];

fs.readdirSync(baseDir).forEach(folder => {
  const folderPath = path.join(baseDir, folder);

  if (fs.lstatSync(folderPath).isDirectory()) {
    fs.readdirSync(folderPath).forEach(file => {
      if (!file.endsWith(".txt")) return;

      const filePath = path.join(folderPath, file);
      const content = fs.readFileSync(filePath, "utf-8").split("\n");

      let record = {};

      // Parse key-value pairs
      content.forEach(line => {
        const [key, value] = line.split(",");
        if (key && value) {
          record[key.trim()] = value.trim();
        }
      });

      // Extract timestamp from filename
      const match = file.match(/weather-data-(\d{4}-\d{2}-\d{2})_(\d{2})-(\d{2})-(\d{2})/);

      if (match) {
        const [_, date, hh, mm, ss] = match;
        record.time = new Date(`${date}T${hh}:${mm}:${ss}`).toISOString();
      }

      // Remove redundant field
      delete record.observation_time;

      // Convert numeric values
      for (let key in record) {
        if (!isNaN(record[key])) {
          record[key] = Number(record[key]);
        }
      }

      output.push(record);
    });
  }
});

// Sort chronologically
output.sort((a, b) => new Date(a.time) - new Date(b.time));

// Save
fs.writeFileSync("data.json", JSON.stringify(output, null, 2));

File: locationcomparisons/data_to_json.py
import os
from datetime import datetime
import json

ORIGDIR = os.getcwd()
RTDIR = os.getcwd() + "/data_files"
os.chdir(RTDIR)

FILENAME = "data.json"
output = []

print(sorted(os.listdir()))
for i in os.listdir():
    folder = f'{RTDIR}/{i}'
    if os.path.isdir(folder):
        os.chdir(folder)
        for file in os.listdir():
            if file[-4:] == '.txt':
                record = {}
                with open(file) as f:
                    content = f.read()
                for line in content.split('\n'):
                    key,value = line.split(",")
                    if key and value:
                        record[key.strip()] = value.strip()
                record.pop('observation_time')
                record['time'] = datetime.strptime(file[13:32], "%Y-%m-%d_%H-%M-%S").isoformat()
                
                for key, value in record.copy().items():
                    try: record[key] = int(value)
                    except (KeyError, ValueError):
                        try: record[key] = float(value)
                        except (KeyError, ValueError): pass

                output.append(record)
output = sorted(output, key=lambda x: x['time'])

os.chdir(ORIGDIR)
print(json.dumps(output))
with open(FILENAME, 'w') as f:
    json.dump(output, f, indent=4)